### DISCUSSION TITLE
What is cargo used for in genmon?

### DISCUSSION THREAD

**Discussion by skipfire** [jgyates/genmon#1146] [Questions] [answered]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open issues](https://github.com/jgyates/genmon/issues),[closed issues](https://github.com/jgyates/genmon/issues?q=is%3Aissue+is%3Aclosed),[open discussion](https://github.com/jgyates/genmon/discussions), [closed discussion](https://github.com/jgyates/genmon/discussions/categories/questions?discussions_q=is%3Aclosed) - which may answer my question or that I could contribute my information to.
- [X] I have read the [Known Issues Wiki Page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and my question / issue is not listed.
- [X] I have provided the version of the software and the operating system in my post.
- [X] If I have an issue with genmon not communicating with the controller, I have read and followed the [serial troubleshooting guide](https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting).

### Version

1.19.03

### Have you submitted your log files from the About page? Note that outbound email must be working for logs to be submitted.

I have not submitted logs

### What is/are your question(s)?

What is the cargo library used for?  I am testing getting another application running on my Pi with GenMon on a Pi 5, but one of the dependencies installed by Cargo (llvm-14-tools) is causing it to fail, so I'm curious what I might break if I remove the library.  It all works fine on Pi 3, 4, and Zero 2.

**Comment by jgyates** [comment]
genmon does not explicitly install cargo. I thought it may be installed as a dependency of one of the libaries so I tried the 'pip3 freeze' command on a pi 4 and pi5 and I do not see this library installed on my test pis. Are you installing any other software on the pi5?

**Reply by skipfire** [reply]
I saw cargo in the genmon install:
https://github.com/jgyates/genmon/blob/fe662a4ee625db43fcc4f22f16889b1c9e1cc8f1/genmonmaint.sh#L183

**Reply by jgyates** [reply, marked as answer]
ok. I forgot about this. I was only checking the phthon libaries used by genmon directly, which are installed via pip.

the line in the instal script that uses apt-get which installs cargo is there to support the install of some libraries that need to be built on the fly instead of downloading pre-built executables for the target processor. If memory serves this was added when there were some incompatibilities with the cryptography python module when using HTTPS with the web interface.

It is possible that this is not needed at this point as incompatibilities that come up between libraries are often ironed out over time by the library maintainers. The short answer is Genmon does not need cargo to operate, although it **might** still be needed during the install process.

**Reply by skipfire** [reply]
Great, then that sounds like it is probably ok to remove afterwards to make things work.  I appreciate the info.  The other end of this problem was with .NET and I've submitted a bug report there about cargo breaking one of the .NET libraries.
